code/modules/assembly/health.dm

```diff
@@ -46,22 +46,26 @@
 		to_chat(user, span_notice("You toggle [src] to \"detect critical state\" mode."))
 
 /obj/item/assembly/health/process()
+	//no ready yet
 	if(!scanning || !secured)
 		return
 
+	//look for a mob in either our location or in the connected holder
 	var/atom/object = src
 	if(connected?.holder)
 		object = connected.holder
-	for(object, object && !ismob(object), object=object.loc);
-	// like get_turf(), but for mobs.
-	var/mob/living/target_mob = object
+	while(!ismob(object))
+		object = object.loc
+		if(isnull(object)) //we went too far
+			return
```

Comment:
```suggestion
			break
```

Reply:
No like but this is more optimal. If we can't find a mob we don't want to proceed

Reply:
In the next line i am converting the object to a mob, rather than checking for null there we can't return here and end our work early, that was the idea

Reply:
oh I see. I mean I guess sure
optimality isn't really important at all but this is nicer